packages/server/src/lib/cache.ts

```diff
@@ -1,69 +1,22 @@
-import { logWarn } from '../utils/logging';
+import memoize from 'memoizee';
 
-type Cache = Record<string, unknown>;
-const cache: Cache = {};
+type CacheableFunc<T> = () => Promise<T>;
+type CachedFunc<T> = CacheableFunc<T> & { clear: () => void };
 
-const retryIntervalMs = 20 * 1000;
+interface CacheOptions {
+    ttlSec: number;
+    warm?: boolean;
+}
```

Comment:
I thought it was slightly nicer to have these arguments in a options object, so that calls that used to look like

```ts
cacheAsync(fn, 60, true);
```

now look like

```ts
cacheAsync(fn, { ttlSec: 60, warm: true });
```